src/backend/catalog/objectaddress.c

```diff
@@ -1011,7 +1011,7 @@ get_object_address(ObjectType objtype, Node *object,
 													   &relation, missing_ok);
 				break;
 			case OBJECT_TRIGGER:
-				if(get_trigger_object_address_hook){
+				if(get_trigger_object_address_hook && sql_dialect == SQL_DIALECT_TSQL){
```

Comment:
If I understand this correctly, `pltsql_protocol_plugin_ptr` is not directly available in the engine to check the `is_tds_client` flag. Maybe it would be better to allow the hook to run and then check its result and run `get_object_address_relobject` if the hook result is empty? I've updated the patch with this change.

Reply:
> Maybe it would be better to allow the hook to run

This apparently breaks `drop trigger if exists` on PG connection, I've reverted this change.

About the manual set of dialect, I can confirm that the following on PG connection:

```sql
set babelfishpg_tsql.sql_dialect = 'TSQL';
comment on trigger trig1 on test1 is 'hello1';
```
will crash the server [inside the hook](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/195044dcde070f6267cd80c6ce50f7b99bde29b0/contrib/babelfishpg_tsql/src/hooks.c#L2342) because `cur_physical_schema` is not available.

To fix that I've changed the check from dialect check to `MyProcPort->is_tds_conn` check.

Reply:
One of the python DDL tests in [github workflow run](https://github.com/babelfish-for-postgresql/babelfish_extensions/actions/runs/8210026056/job/22456716888?pr=2414) seems to crash the server with:

```
2024-03-08T23:08:45.1841383Z +~~ERROR (Code: 10054)~~
2024-03-08T23:08:45.1842030Z +~~ERROR (Message: [08S01] [Microsoft][ODBC Driver 17 for SQL Server]TCP Provider: Error code 0x2746 (10054) (SQLExecDirectW))~~
2024-03-08T23:08:45.1843078Z +~~ERROR (Code: 0)~~
2024-03-08T23:08:45.1843659Z +~~ERROR (Message: [08S01] [Microsoft][ODBC Driver 17 for SQL Server]Communication link failure (0) (SQLExecDirectW))~~
2024-03-08T23:08:45.1844477Z +~~ERROR (Code: 0)~~
2024-03-08T23:08:45.1845053Z +~~ERROR (Message: [08S01] [Microsoft][ODBC Driver 17 for SQL Server]Communication link failure (0) (SQLExecDirectW))~~
...
```

It is hard to tell which one, the output there is not linear (though triggers only present in `ddl_all_objects` and `dld_triggers` tests). Just I cannot reproduce it locally, all `ddl_` python tests are running to completion for me (both with or without sqltoolsservice setup for DDL export). I will re-run GitHub tests once more before further investigation.

Reply:
Tests have passed on 4-th re-run without any changes, seems to be some kind on intermittent issue (OOM on runner?) that happened 3 times in a row. I've added [engine revision output after the checkout](https://github.com/babelfish-for-postgresql/babelfish_extensions/actions/runs/8217942911/job/22474194575?pr=2414#step:4:92) to be sure that the last commit from `comment_on_pg_trigger` branch was used in testing.

Reply:
Let's keep dialect check and TDS conn check to be safer side.

Reply:
Added the dialect check in addition to the connection type check.